### PR TITLE
Update readme for actions and guards

### PR DESCRIPTION
**Pull request by adamkiss**
So, I updated the readme. Additionally, there's a repo at [adamkiss/kirby-uniform-demonstration](https://github.com/adamkiss/kirby-uniform-demonstration) with a demonstration of a clean 3.5.7.1 starterkit with uniform, custom guard and a custom actions.

I've seen the note in the #214 to make it AJAX compatible, but I've tried to do it with "default Javascript" and haven't gotten anywhere - I think it's highly dependent on:

1. what you're actually trying to achieve (content representations/routes/etc.)
2. your build process, your workflow and what libraries you use (and how you use them).

So that's not part of this PR.

---

The uniform demonstration repo will stay alive, so you can link it if you want, or we can transfer it to you, or I can give you access and you can try adding the XHR submission.

### CODE REVIEW

**Review by mzur — CHANGES_REQUESTED** [review]
That's not enough, unfortunately. We need an `index.php` that calls the `load()` helper on all custom action/guard classes and the action/guard classes in new separate files as [described in my comment](https://github.com/mzur/kirby-uniform/issues/214#issuecomment-915487272). Could you please add these changes, too?

**Comment by adamkiss** [comment]
Interesting. Is that specifically for non-composer installs? I can try a non-composer install, and see what’s the happy minimal path

**Comment by mzur** [comment]
Yes, that's required for non-Composer installs but should work with Composer too.

**Comment by adamkiss** [comment]
@mzur Finally, after half year, I got around to finishing to this :D

I expanded (and tested on a pure [starterkit demo install](https://github.com/adamkiss/kirby-uniform-demonstration/tree/starterkit)) the section about putting the custom actions/guards into a separate file and autoloading it via Kirby at a correct time.

Of course, this _might_ me simplified if you're using composer, but I wasn't sure how to note that down in the documentation - for the composer setup, you don't even have to add the plugin definition - just an `index.php` with correct namespace and your action/guard classes and you're done.

File: docs/guards/guards.md [hunk 1]
```diff
@@ -55,9 +55,9 @@ Although you probably want to use magic methods in most cases, it might be usefu
 
 ## Custom Guards
 
-As you have seen in the previous section, each guard is a class in the `Uniform\Guards` namespace. To write your own guards you can either provide them through a PHP package and Composer or implement them in the traditional way as a Kirby plugin. The plugin may be a file `site/plugins/uniform-guards.php` that will be automatically loaded by Kirby.
+As you have seen in the previous section, each guard is a class in the `Uniform\Guards` namespace. To write your own guards you can either provide them through a PHP package and Composer or implement them in the traditional way as a Kirby plugin. The plugin may be a file `site/plugins/custom-guards/index.php` that will be automatically loaded by Kirby.
```
Comment (eXpl0it3r) [inline docs/guards/guards.md:58]: Maybe `site/plugins/uniform-custom-guards` instead?

File: docs/actions/actions.md [hunk 2]
```diff
@@ -44,30 +44,51 @@ if (kirby()->request()->is('POST')) {
 
 ## Custom Actions
 
-Custom actions are very similar to [custom guards](/guards/guards#custom-guards), too. Each action is a class in the `Uniform\Actions` namespace. To write your own actions you can either provide them through a PHP package and Composer or implement them in the traditional way as a Kirby plugin. The plugin may be a file `site/plugins/uniform-actions.php` that will be automatically loaded by Kirby.
+Custom actions are very similar to [custom guards](/guards/guards#custom-guards), too. Each action is a class in the `Uniform\Actions` namespace. To write your own actions you can either provide them through a PHP package and Composer or implement them in the traditional way as a Kirby plugin.
```
Comment (eXpl0it3r) [inline docs/actions/actions.md:47]: You removed the last sentence here, but left it for the guard part. Might be better to keep it in sync

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
I think there's still room from a usability point of view, to communicate not just how, but why things are needed and maybe some potential limits, but I think beyond the minor things commented, it's good and will help clear up some confusion 😃

File: docs/guards/guards.md (hunk 1 above)
Comment (adamkiss) [reply docs/guards/guards.md:58]: This sentence is wrong - we can't make it easily load Uniform namespace without Composer. It's gone

File: docs/actions/actions.md (hunk 2 above)
Comment (adamkiss) [reply docs/actions/actions.md:47]: Removed both

**Comment by adamkiss** [comment]
@eXpl0it3r Thank's for the review - I removed the sentence about one `index.php` being enough, since we can't tell it to properly autoload stuff in non-composer setup, without depending on the Kirby plugin load order. If the Uniform is loaded first (like having `kirby-uniform` and `uniform-custom-xxx` plugins), it works without the load helper, and the custom guards and actions can just be defined in the `index.php` of the plugin. Otherwise, we need to manually load the the class file from `index.php` into the proper namespace.

@mzur I thought about including that ^ part to the docs as well, but wasn't quite sure how to put it.

**Review by eXpl0it3r — APPROVED** [review]
I'd probably still incorporate some of the explanation you've written here, at the same time, I'd rather have this merged now, than later. 😄

**Comment by adamkiss** [comment]
@eXpl0it3r It's waiting on @mzur - they're probably busy 🤷‍♂️

**Comment by mzur** [comment]
Yes, sorry, I'll get to it eventually.

**Review by mzur — APPROVED** [review]
Sorry for the long wait!

@adamkiss Looks good. I [removed](https://github.com/mzur/kirby-uniform/pull/217/commits/ff58f754a5b8a9eb962da66f9c8f5feb76e9ff8c#diff-e9b5ec98255aa641dd2d80990fd4dd12989ba92335173eb737db235452ceff2dL88) the calls to `Kirby::plugin()`. Could you please confirm if they are not required (I think not)?

**Comment by mzur** [comment]
Thanks!

**Pull request merged**